### ISSUE TITLE
Is coursera-dl able to save webpages?

### ISSUE THREAD

**Issue by shixudongleo** [coursera-dl/coursera-dl#356]
When I use coursera-dl, I am only able to save video, slides, and other txt files. 
However, for coursera archive, I also want to save some key web pages, such as syllabus page,  homework page, etc. 

Is coursera-dl abel to save web pages now?
If coursera-dl does not support saving web pages currently, I am thinking of add this feature. 
Can you give some instructions? 

Thank you.

**Comment by rbrito** [comment]
@shixudongleo, no, those pages are not yet saved. If you can add that feature, I'd love to merge your work into the main program.

It *seems* to me that the syllabus of the courses (and similar pages) are found at places like;

* https://class.coursera.org/algo2-004/wiki/syllabus
* https://class.coursera.org/posacommunication-001/wiki/syllabus

But I am really, really sure that instructors break this a lot when they deploy their courses on coursera, as they don't like to follow the general structure of the site. :(

Perhaps the best thing would be to parse the initial page of each course, look for a link called syllabus on the left side and consider that as the link to the syllabus.

I hope that this helps with your implementation.

Regards,

Rogério Brito.

But, by all means, go ahead and let us

**Comment by rbrito** [comment]
OK, I'm conviced that, at least, the wiki pages would be nice to have locally and that's a big feature request.

`wget` is able to mirror sites (I usually use it the option `--mirror --no-parent`) and we would either:

* need to emulate it via Python
* or use it potentially passing some information like cookies/etc as headers.

I'm not really sure what would be the appropriate thing to do here.

Having a subdirectory named something like `00-Course Wiki` in the respective course's directory would be great if we decide to go that way.


Thanks,

Rogério.

**Comment by balta2ar** [comment]
@shixudongleo FYI, recent commits allowed to save text of most of programming assignments and supplement sections in on-demand courses (including images in those texts).

**Comment by wadkar** [comment]
@balta2ar ,
Could you please explain how to do it? Current README.md (front page) does not allude to it. I can help test/debug it off master and even submit PR for fixes if I can find time.

**Comment by balta2ar** [comment]
@wadkar This is only implemented for on-demand courses. Text instructions that you can find in supplement materials/programming assignments are now saved into html files. You don't need to do anything special, just run the latest version and instructions will be saved as well.

Just to be clear, this isn't about old-style courses. Only on-demand courses.

Yes, README is not updated yet. Stay tuned.

**Comment by balta2ar** [comment]
related #490

**Comment by balta2ar** [comment]
@shixudongleo Is this still actual?
